# [M2-TECHNICAL] Technical Assessment 2
Ralph Cajipe

The ABC Telcom Customers dataset

In [1]:
import pandas as pd

# 1. Create

### 1.1 Create from a XLSX

The ABC Telcom Customers dataset

In [2]:
# Importing the openpyxl module to read and write excel files since we will use a .xls file
import openpyxl

df = pd.read_excel('clean-CustomerInfo.xlsx')

# 2. Read

The reading should be a long process to ensure you fully understand the data.

### 2.1 Show the shape of the data

In [3]:
df.shape

(600, 8)

### 2.2 Show Top 5 and Bottom 5 Rows

In [4]:
df.head(10)

Customer_ID            Occupation PROVINCE  GENDER MARITAL_STATUS URBANITY  \
0   009-17336   White collar worker      TK1  FEMALE        MARRIED    Rural   
1   009-17170         Small Farrmer       TK  FEMALE        MARRIED    Rural   
2   009-17046        Factory worker       TK  FEMALE        MARRIED    Rural   
3   009-17381   Construction Worker       TK  FEMALE        WIDOWER    Rural   
4   009-17088        Factory worker       TK  FEMALE        WIDOWER    Rural   
5   009-17186  Small Business Owner       TK  FEMALE        MARRIED    Rural   
6   009-17283        Factory worker       TK    MALE        MARRIED    Rural   
7   009-17058        Factory worker       TK  FEMALE         SINGLE    Rural   
8   009-17535          Small Farmer       TK  FEMALE        MARRIED    Rural   
9   009-17140         Small Farrmer    Takeo  FEMALE        MARRIED    Rural   

  CREDIT_BUREAU  INCOME_TIER  
0             B          2.0  
1             D          2.0  
2             D          2.0  
3             D          2.0  
4             D          3.0  
5             D          3.0  
6           NaN          3.0  
7             A          2.0  
8             D          2.0  
9           NaN          2.0

In [5]:
df.tail()

Customer_ID           Occupation PROVINCE  GENDER MARITAL_STATUS URBANITY  \
595   009-17211       Factory worker       TK    MALE        MARRIED    Rural   
596   009-17593                  NaN       TK  FEMALE        WIDOWER    Rural   
597   009-17279         Small Farmer       TK  FEMALE        WIDOWER    Rural   
598   009-17433       Factory worker       KS    MALE        MARRIED    Rural   
599   009-17396  White collar worker       TK    MALE        MARRIED    Rural   

    CREDIT_BUREAU  INCOME_TIER  
595             A          3.0  
596           NaN          NaN  
597             D          2.0  
598             B          3.0  
599             B          3.0

### 2.3 Show Columns and Data Type

In [7]:
df.columns

Index(['Customer_ID', 'Occupation', 'PROVINCE', 'GENDER', 'MARITAL_STATUS',
       'URBANITY', 'CREDIT_BUREAU', 'INCOME_TIER'],
      dtype='object')

In [8]:
df.dtypes

Customer_ID        object
Occupation         object
PROVINCE           object
GENDER             object
MARITAL_STATUS     object
URBANITY           object
CREDIT_BUREAU      object
INCOME_TIER       float64
dtype: object

### 2.4 Summary Statistics

In [9]:
df.describe()

INCOME_TIER
count   592.000000
mean      2.807432
std       0.818615
min       1.000000
25%       2.000000
50%       3.000000
75%       3.000000
max       5.000000

In [10]:
# This is a function that will return the number of unique values in a column
df.describe(include='object')

Customer_ID      Occupation PROVINCE  GENDER MARITAL_STATUS URBANITY  \
count          600             592      600     600            600      600   
unique         600              12        9       2              4        1   
top      009-17336  Factory worker       TK  FEMALE        MARRIED    Rural   
freq             1             176      558     452            412      600   

       CREDIT_BUREAU  
count            440  
unique             4  
top                D  
freq             152

### 2.5 Filtering columns

In [11]:
df.Occupation

0      White collar worker
1            Small Farrmer
2           Factory worker
3      Construction Worker
4           Factory worker
              ...         
595         Factory worker
596                    NaN
597           Small Farmer
598         Factory worker
599    White collar worker
Name: Occupation, Length: 600, dtype: object

### 2.6 Filtering on rows

In [12]:
df.head()

Customer_ID           Occupation PROVINCE  GENDER MARITAL_STATUS URBANITY  \
0   009-17336  White collar worker      TK1  FEMALE        MARRIED    Rural   
1   009-17170        Small Farrmer       TK  FEMALE        MARRIED    Rural   
2   009-17046       Factory worker       TK  FEMALE        MARRIED    Rural   
3   009-17381  Construction Worker       TK  FEMALE        WIDOWER    Rural   
4   009-17088       Factory worker       TK  FEMALE        WIDOWER    Rural   

  CREDIT_BUREAU  INCOME_TIER  
0             B          2.0  
1             D          2.0  
2             D          2.0  
3             D          2.0  
4             D          3.0

# 3. Update/Clean

### 3.1 We will not drop missing values in an instance someone happens to have no occupation, a customer with no occupation is still a customer and among other reasons can be used for analysis.

### 3.2 Drop duplicates

In [ ]:
# Drop duplicates
df.drop_duplicates()

In [14]:
# Check for duplicates
df.duplicated().sum()

0

### 3.3 Fix mispelled features or column names

In [18]:
# Rename "Small Farrmer" to "Small Farmer"
df['Occupation'] = df['Occupation'].replace('Small Farrmer', 'Small Farmer')

### 3.4 Fix mispelled values in a column

We are dealing with provinces in Cambodia, so we will fix the mispelled values in the province column.

In [34]:
# In PROVINCE column, replace "TK" and "TK1", with "Takéo" 
df['PROVINCE'] = df['PROVINCE'].replace(['TK', 'TK1'], 'Takéo')

In [37]:
# In PROVINCE column, replace "BM" with "Banteay Meanchey"
df['PROVINCE'] = df['PROVINCE'].replace('BM', 'Banteay Meanchey')

In [40]:
# In PROVINCE column, replace "KS" with "Kampong Speu"
df['PROVINCE'] = df['PROVINCE'].replace('KS', 'Kampong Speu')

In [42]:
# In PROVINCE column, replace "KP" with "Kampot"
df['PROVINCE'] = df['PROVINCE'].replace('KP', 'Kampot')

In [45]:
# Show all provinces with fixed names
df['PROVINCE'].value_counts()

Takéo               575
Banteay Meanchey      9
Kampong Speu          8
Kampot                8
Name: PROVINCE, dtype: int64

In [ ]:
# Show the entire dataframe with no ellipsis in the middle so we can see everything
pd.set_option('display.max_rows', None)
df


# 4. Output

### 4.1 Output to CSV

In [48]:
df.to_csv('output-CustomerInfo.csv')

In [49]:
df.to_html('output-CustomerInfo.html')

# 5. Customer Analysis

### 5.1 Find the most common occupation among customers

In [57]:
# Find out what occupation the majority of customers are in
df['Occupation'].value_counts()


Small Farmer                180
Factory worker              176
Construction Worker          68
White collar worker          40
SME Farmer                   36
Food supplier/restaurant     24
Other                        20
Small Business Owner         16
Car Repair Shop              16
Medium Biz                   12
Land Transport Service        4
Name: Occupation, dtype: int64

The occupation that the majority of customers are in is "Small Farmer" with 180 customers. 
The least number of customers is "Land Transportation Service" with 4 customers.

### 5.2 Count the total number of unique customers

In [52]:
df['Customer_ID'].nunique()

600

In [53]:
df.describe(include='object')

Customer_ID    Occupation PROVINCE  GENDER MARITAL_STATUS URBANITY  \
count          600           592      600     600            600      600   
unique         600            11        4       2              4        1   
top      009-17336  Small Farmer    Takéo  FEMALE        MARRIED    Rural   
freq             1           180      575     452            412      600   

       CREDIT_BUREAU  
count            440  
unique             4  
top                D  
freq             152

### 5.3 Find customers with no occupation

In [54]:
df[df['Occupation'].isnull()]

Customer_ID Occupation PROVINCE  GENDER MARITAL_STATUS URBANITY  \
55    009-17010        NaN    Takéo  FEMALE        WIDOWER    Rural   
84    009-17310        NaN    Takéo  FEMALE        WIDOWER    Rural   
127   009-17161        NaN    Takéo  FEMALE        WIDOWER    Rural   
204   009-17442        NaN    Takéo  FEMALE        WIDOWER    Rural   
233   009-17293        NaN    Takéo  FEMALE        WIDOWER    Rural   
285   009-17142        NaN    Takéo  FEMALE        WIDOWER    Rural   
313   009-17461        NaN    Takéo  FEMALE        WIDOWER    Rural   
596   009-17593        NaN    Takéo  FEMALE        WIDOWER    Rural   

    CREDIT_BUREAU  INCOME_TIER  
55            NaN          NaN  
84            NaN          NaN  
127           NaN          NaN  
204           NaN          NaN  
233           NaN          NaN  
285           NaN          NaN  
313           NaN          NaN  
596           NaN          NaN

In [55]:
# Count customers with no occupation
df['Occupation'].isnull().sum()

8

### 5.4 Find out from which province the majority of customers are from

In [56]:
df['PROVINCE'].value_counts()

Takéo               575
Banteay Meanchey      9
Kampong Speu          8
Kampot                8
Name: PROVINCE, dtype: int64

The majority of customers come from Takéo Province with 575 customers, while others come from Banteay Meanchey, Kampong Speu, and Kampot.

### 5.5 In GENDER column, find out how many customers are male and female

In [58]:
df['GENDER'].value_counts()

FEMALE    452
MALE      148
Name: GENDER, dtype: int64

Remember, this is the total count of workers for each occupation.

In [63]:
df['Occupation'].value_counts()

Small Farmer                180
Factory worker              176
Construction Worker          68
White collar worker          40
SME Farmer                   36
Food supplier/restaurant     24
Other                        20
Small Business Owner         16
Car Repair Shop              16
Medium Biz                   12
Land Transport Service        4
Name: Occupation, dtype: int64

### 5.6 Which occupations are most common among male and female?

In [62]:
df.groupby(["Occupation", "GENDER"]).size().sort_values(ascending=False)


Occupation                GENDER
Small Farmer              FEMALE    144
Factory worker            FEMALE    140
Construction Worker       FEMALE     68
Factory worker            MALE       36
Small Farmer              MALE       36
White collar worker       MALE       24
SME Farmer                FEMALE     24
White collar worker       FEMALE     16
SME Farmer                MALE       12
Food supplier/restaurant  FEMALE     12
                          MALE       12
Small Business Owner      FEMALE     12
Other                     MALE       12
Car Repair Shop           FEMALE      8
                          MALE        8
Medium Biz                FEMALE      8
Other                     FEMALE      8
Small Business Owner      MALE        4
Medium Biz                MALE        4
Land Transport Service    FEMALE      4
dtype: int64

### 5.7 In MARITAL_STATUS, how many are single, married, divorced, or widowed?

In [64]:
df['MARITAL_STATUS'].value_counts()

MARRIED     412
WIDOWER     172
SINGLE        8
DIVORCED      8
Name: MARITAL_STATUS, dtype: int64

### 5.8 What are the different values in URBANITY?

In [66]:
df['URBANITY'].value_counts()

Rural    600
Name: URBANITY, dtype: int64

Rural is the only type of urbanity, so it makes sense now that the occupations are connected to agricultural areas and areas with forestry in the provinces of Cambodia.

### 5.9 What are the different CREDIT_BUREAU values?

In [69]:
df['CREDIT_BUREAU'].value_counts()

D    152
B    100
A     96
C     92
Name: CREDIT_BUREAU, dtype: int64

The CREDIT_BUREAU column likely contains information on an individual's credit score from a credit bureau. This information could be used to determine an individual's creditworthiness or to assess the risk of lending money to an individual.

For the CREDIT_BUREAU column, A is the highest credit score, B is the second highest, C is the third highest, and D is the lowest. 

### 5.10 What are the different INCOME_TIER values?

In [70]:
df['INCOME_TIER'].value_counts()

3.0    282
2.0    206
4.0     68
5.0     24
1.0     12
Name: INCOME_TIER, dtype: int64

For the INCOME_TIER column, 1 is the lowest income, 2 is the second lowest, 3 is the third lowest, 4 is the fourth lowest, and 5 is the highest income.

There is a moderate positive correlation between the CREDIT_BUREAU and INCOME_TIER columns. This means that, generally, individuals who have a higher credit score from a credit bureau also tend to have a higher income.

# CONCLUSION

The Customer_ID, Occupation, Province, Gender, MARITAL_STATUS, URBANITY, CREDIT_BUREAU, and INCOME_TIER columns can be used to segment the dataset by different characteristics. For example, you could look at the average credit score from a credit bureau for individuals in different occupations, provinces, genders, etc. But in the case of the ABC Telcom Customers dataset, the Occupation column is the most useful for segmentation because it has the most unique values.

There is a moderate positive correlation between the INCOME_TIER and CREDIT_BUREAU columns. This means that, generally, individuals who have a higher income also tend to have a higher credit score from a credit bureau. 